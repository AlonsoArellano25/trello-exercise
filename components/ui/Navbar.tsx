import { AppBar, Link, Toolbar, Typography } from "@mui/material";
import React from "react";
import NextLink from "next/link";

export const Navbar = () => {
  return (
    <AppBar position="sticky">
      <Toolbar data-testid="toolbar">
        <NextLink href={"/"} passHref>
          <Link underline="none" color={"white"}>
            <Typography data-testid="title" variant="h6">
              Trello
            </Typography>
          </Link>
        </NextLink>
      </Toolbar>
    </AppBar>
  );
};
