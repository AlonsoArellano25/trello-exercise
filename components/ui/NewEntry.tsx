import { Box, Button, TextField } from "@mui/material";
import SaveOutlinedIcon from "@mui/icons-material/SaveOutlined";
import AddCircleOutlineOutlinedIcon from "@mui/icons-material/AddCircleOutlineOutlined";
import { ChangeEvent, useContext, useState } from "react";
import { EntriesContext } from "../../context/entries";
import { UIContext } from "../../context/ui";

export const NewEntry = () => {
  const [inputValue, setInputValue] = useState("");
  const [touched, setTouched] = useState(false);
  const { addNewEntry } = useContext(EntriesContext);
  const { isAddingEntry, setIsAddingEntry } = useContext(UIContext);

  const onTextFieldChange = (event: ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  };

  const onSave = () => {
    if (inputValue.length === 0) return;
    addNewEntry(inputValue);
    setIsAddingEntry(false);
    setTouched(false);
    setInputValue("");
  };
  return (
    <Box sx={{ marginBottom: 2, paddingX: 1 }}>
      {isAddingEntry ? (
        <>
          <TextField
            data-testid="textfield-new-entry"
            fullWidth
            sx={{ marginTop: 2, marginBottom: 1 }}
            placeholder="New entry"
            autoFocus
            multiline
            label="New entry"
            helperText={inputValue.length <= 0 && touched && "Enter a value"}
            error={inputValue.length <= 0 && touched}
            value={inputValue}
            onChange={onTextFieldChange}
            onBlur={() => setTouched(true)}
          />
          <Box display="flex" justifyContent="space-between">
            <Button
              data-testid="button-cancel"
              variant="outlined"
              onClick={() => setIsAddingEntry(false)}
            >
              Cancel
            </Button>
            <Button
              data-testid="button-save"
              variant="outlined"
              color="secondary"
              endIcon={<SaveOutlinedIcon />}
              onClick={onSave}
            >
              Save
            </Button>
          </Box>
        </>
      ) : (
        <Button
          data-testid="button-add-task"
          startIcon={<AddCircleOutlineOutlinedIcon />}
          fullWidth
          variant="outlined"
          onClick={() => setIsAddingEntry(true)}
        >
          Add Task
        </Button>
      )}
    </Box>
  );
};
