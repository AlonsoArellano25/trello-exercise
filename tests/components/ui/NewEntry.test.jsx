import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { NewEntry } from "../../../components/ui";

describe("Tests in  <NewEntry />", () => {
  it("should render properly button cancel", () => {
    render(<NewEntry />);
    const bElement = screen.getByTestId("button-add-task");
    expect(bElement).toHaveTextContent("Add Task");
  });
  it("should match with snapshot", () => {
    const { container } = render(<NewEntry />);
    expect(container).toMatchSnapshot();
  });
  // it("should render properly text field", () => {
  //   render(<NewEntry />);
  //   const bElement = screen.getByTestId("textfield-new-entry");
  //   expect(bElement).toHaveValue("");
  // });
});
