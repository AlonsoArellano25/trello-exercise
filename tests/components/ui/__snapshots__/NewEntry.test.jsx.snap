// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in  <NewEntry /> should match with snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-8ziurz"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-xsqe0f-MuiButtonBase-root-MuiButton-root"
      data-testid="button-add-task"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="AddCircleOutlineOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      Add Task
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
